wagtail/contrib/redirects/views.py

```diff
@@ -1,23 +1,23 @@
 import os
 
 from django.core.exceptions import PermissionDenied, SuspiciousOperation
-from django.core.paginator import InvalidPage, Paginator
 from django.db import transaction
-from django.db.models import Q
-from django.http import Http404
 from django.shortcuts import get_object_or_404, redirect, render
 from django.template.response import TemplateResponse
 from django.urls import reverse
 from django.utils.encoding import force_str
+from django.utils.functional import cached_property
 from django.utils.translation import gettext as _
-from django.utils.translation import ngettext
+from django.utils.translation import gettext_lazy, ngettext
 from django.views.decorators.http import require_http_methods
-from django.views.decorators.vary import vary_on_headers
 
 from wagtail.admin import messages
 from wagtail.admin.auth import PermissionPolicyChecker
 from wagtail.admin.forms.search import SearchForm
+from wagtail.admin.ui.tables import Column, StatusTagColumn, TitleColumn
+from wagtail.admin.views.generic import IndexView
```

Comment:
So we can use the name `IndexView` instead of `Index`

```suggestion
from wagtail.admin.views import generic
```